packages/decoder/lib/types.ts

```diff
@@ -125,11 +125,28 @@ export interface ContractState {
 
 /**
  * This type represents a web3 Log object that has been decoded.
- * Note that it extends the Log type and just adds an additional field
- * with the decoding.
+ * Note that it extends the Log type.   The differences are:
+ *
+ *   1. It adds an additional field with the decodings;
+ *   2. Some fields that in Log are allowed to be strings,
+ *     here are specified as numbers.  (Because this type is
+ *     for output and that type is for input.)
  * @category Results
  */
 export interface DecodedLog extends Log {
+  /**
+   * Index of the log within the block.
+   */
+  logIndex?: number;
+  /**
+   * Index within the block of the emitting transaction; null if
+   * block is pending.
+   */
+  transactionIndex?: number | null;
```

Comment:
I think these optional fields need explanation on why they might be `undefined`. Probably warrants a note in the `DecodedLog` docstring, to say that @truffle/decoder _re-emits_ everything given, and unspecified optional input fields will thus not appear in output

Reply:
Actually, that's not how this works!  Which raises a good point: These fields shouldn't be optional.  I guess I should just not make this an extension of `Log`, huh?  Because it doesn't save me any work if everything is mandatory. :-/

To explain: If you use `decodeLog`, passing in your own logs, you just get back the decodings, you don't get `DecodedLog`.  You only get `DecodedLog` if you call `events()`, to which one does not pass in their own logs.  So actually these fields will always be present in this type.

...so do you agree that's what I should do?

Reply:
Can this emit the same info in both situations? It's weird that passing in a `Log` will remove fields, while `events()` will return everything log-related

Reply:
What?  Passing in a `Log` doesn't remove fields.  What I'm saying is that `decodeLog` doesn't use the `DecodedLog` type at all.  It returns the decodings, they're not inside a `DecodedLog` container.  Only `events` uses that.

Reply:
Ohh, sorry to misunderstand. Yeah, let's not extend `Log`